### DISCUSSION TITLE
Date formula

### DISCUSSION THREAD

**Discussion by twilson9034** [Budibase/budibase#3729] [Help] [answered]
I'm trying to set up a formula for comparing dates. I have a column for "expire date" and a "due days" column. I'm first trying to test the formula before creating a column with it. Here is the formula, basically if the expire date is within due days (30) of current day Due or not Due.
{{ #lte date (subtract ( date Form.Fields.Expire Date  "x") (multiply 30 86400000))  date now  }} Due {{ else }} Not Due {{/lte}} 

when I use this I just get false in the text field I'm testing in.

**Comment by twilson9034** [comment]
I've used {{#if (gt date Notify Date) }} Due {{else}} Not Due{{/if}}. I can get it to work if I'm using the date pickers themselves but not stored values.

**Comment by mjashanks** [comment, marked as answer]
Hi @twilson9034 

That's some crazy handlebars :) 

This is why we introduced Javascript. Here it is in Javascript:

```javascript
const now = new Date();
const expiry = $(" Form.Fields.Expire Date");

if (!expiry) return "Expiry not set";

const dueDate = new Date(expiry);
dueDate.setDate(dueDate.getDate() + 30);

if (now < dueDate) {
	return "Not Due";
} else {
	return "Due";
}

```

**Reply by twilson9034** [reply]
@mjashanks That worked, Thank you. I'm not well versed in javascript... yet as I see how to use it then ill be able to get it figured out. 

I used those handlebars because that is the only thing I could find in some discussions.

**Comment by Hades1280** [comment]
Hi, I have a similar problem. I'm using a javascript to count the days elapsed between dates, but I need to determine the working(business) days elapsed, not the calendar days.

With these days counted, an alert is triggered when a given limit of days is reached.

How is that possible in budibase (count business days). I appreciate your help. Thanks.

This is my code:

`var date_1 = new Date($("New Repeater.solicitudes.fechaingreso"));
var date_2 = new Date(helpers.date($("now"), "YYYY-MM-DD"));

var day_as_milliseconds = 86400000;
var diff_in_millisenconds = date_2 - date_1;
var diff_in_days = diff_in_millisenconds / day_as_milliseconds;

return diff_in_days + " días desde que fue ingresado."; `
